### PR TITLE
Nanava/configurejs

### PR DESCRIPTION
**Pull request by t-nanava**
**Added InstanceManager in JSI Path**

#### Description

Earlier InstanceManager was working only with the executor path. Now, it has been added to the JSI path for V8 Engine.

#### Validation

The current branch is tested with V8 engine and validated with the help of RNTester application.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/96)

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/Android.mk [hunk 1]
```diff
@@ -12,6 +12,7 @@ LOCAL_MODULE := v8executor
 LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp)
 
 LOCAL_C_INCLUDES := $(LOCAL_PATH)
+LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../../../..
```
[the remark this answers is not in the document]
Comment (t-nanava) [reply ReactAndroid/src/main/java/com/facebook/react/v8executor/Android.mk]: Will it be alright if we use REACT_SRC_DIR from build.gradle?

[the remark this answers is not in the document]
Comment (t-nanava) [reply ReactAndroid/src/main/java/com/facebook/react/v8executor/Android.mk]: Added a new variable REACT_JNI_DIR

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/InstanceManager.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+#include "InstanceManager.h"
+#include <jni/react/jni/ReadableNativeMap.h>
+
+
+namespace facebook {
+namespace react {
+
+namespace {
+
+class V8ExecutorFactory : public JSExecutorFactory {
```
[the remark this answers is not in the document]
Comment (t-nanava) [reply ReactAndroid/src/main/java/com/facebook/react/v8executor/InstanceManager.cpp]: Refactored it into V8ExecutorFactory.h

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/InstanceManager.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+#include <folly/dynamic.h>
+#include <jsiexecutor/jsireact/JSIExecutor.h>
+#include <jsi/V8Runtime.h>
+#include "react/jni/JSLoader.h"
+#include <react/jni/JSLogging.h>
+#include "InstanceManager.h"
+#include <react/jni/ReadableNativeMap.h>
+#include "V8ExecutorFactory.h"
+
+namespace facebook { namespace react { namespace jsi{
```
Comment (rohitjain85) [inline ReactAndroid/src/main/java/com/facebook/react/v8executor/InstanceManager.cpp]: >{ [](start = 52, length = 1)

nit: space b/w jsi annd {

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/Android.mk [hunk 4]
```diff
@@ -12,6 +12,7 @@ LOCAL_MODULE := v8executor
 LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp)
 
 LOCAL_C_INCLUDES := $(LOCAL_PATH)
+LOCAL_C_INCLUDES += $(THIRD_PARTY_NDK_SRC_DIR)/..
```
Comment (rohitjain85) [inline ReactAndroid/src/main/java/com/facebook/react/v8executor/Android.mk]: can we in above like only

LOCAL_C_INCLUDES := $(LOCAL_PATH)  $(THIRD_PARTY_NDK_SRC_DIR)/..

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/OnLoad.cpp [hunk 5]
```diff
@@ -11,40 +11,11 @@
 #include <react/jni/JReactMarker.h>
 #include <react/jni/JSLogging.h>
 #include <react/jni/ReadableNativeMap.h>
+#include "V8ExecutorFactory.h"
```
Comment (rohitjain85) [inline ReactAndroid/src/main/java/com/facebook/react/v8executor/OnLoad.cpp]: nit: preserve the line

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/V8ExecutorFactory.cpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#include <jsi/V8Runtime.h>
+#include "react/jni/JSLoader.h"
+#include <react/jni/JSLogging.h>
+#include <react/jni/ReadableNativeMap.h>
+
+namespace facebook { namespace react { namespace jsi {
+
+std::unique_ptr<JSExecutor> V8ExecutorFactory::createJSExecutor(
+      std::shared_ptr<ExecutorDelegate> delegate,
+      std::shared_ptr<MessageQueueThread> jsQueue){
```
Comment (rohitjain85) [inline ReactAndroid/src/main/java/com/facebook/react/v8executor/V8ExecutorFactory.cpp]: >){ [](start = 49, length = 2)

nit: space b/w ){

File: ReactAndroid/src/main/java/com/facebook/react/v8executor/V8ExecutorFactory.h [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#include <react/jni/JSLogging.h>
+#include <react/jni/ReadableNativeMap.h>
+
+
+namespace facebook { namespace react { namespace jsi {
+
+class V8ExecutorFactory : public JSExecutorFactory {
+public:
+  V8ExecutorFactory(folly::dynamic&& v8Config) :
+    m_v8Config(std::move(v8Config)) {}
```
Comment (rohitjain85) [inline ReactAndroid/src/main/java/com/facebook/react/v8executor/V8ExecutorFactory.h]: >  m_v8Config(std::move(v8Config)) {} [](start = 2, length = 36)

move definition to cpp file

**Pull request merged**